tests/test_logs.py

```diff
@@ -0,0 +1,97 @@
+import os
+import re
+from pathlib import Path
+
+import pytest
+from click.testing import CliRunner
+
+from unfurl.__main__ import detect_log_level, detect_verbose_level
+from unfurl.job import JobOptions, Runner
+from unfurl.localenv import LocalEnv
+from unfurl.logs import Levels
+
+
+def test_format_of_job_file_log():
+    cli_runner = CliRunner()
+    with cli_runner.isolated_filesystem():
+        path = Path(__file__).parent / "examples" / "shell-ensemble.yaml"
+        with open(path) as f:
+            ensemble = f.read()
+        with open("ensemble.yaml", "w") as f:
+            f.write(ensemble)
+        manifest = LocalEnv().get_manifest()
+        runner = Runner(manifest)
+
+        runner.run(JobOptions(instance="test1"))
+        log_file = list((Path.cwd() / "jobs").glob("*.log"))[0]
+
+        with open(log_file) as f:
+            first_line = f.readline()
+            # check format - it should differ from console log
+            expected_format = r"\[.+\] unfurl:INFO: starting deploy job for .*"
+            assert re.match(expected_format, first_line)
+
+
+class TestLogLevelDetection:
+    def test_default(self):
+        level = detect_log_level(loglevel=None, quiet=False, verbose=0)
+        assert level is Levels.INFO
+
+    def test_quiet_mode(self):
+        level = detect_log_level(loglevel=None, quiet=True, verbose=0)
+        assert level is Levels.CRITICAL
+
+    def test_from_env_var(self):
+        try:
+            os.environ["UNFURL_LOGGING"] = "DEBUG"
+            level = detect_log_level(loglevel=None, quiet=False, verbose=0)
+            assert level is Levels.DEBUG
+        finally:
+            del os.environ["UNFURL_LOGGING"]
```

Comment:
don't you need to save and restore UNFURL_LOGGING?

Reply:
I set it to difference levels all the time when I'm running unit tests -- just save and restore environment variable in a try finally block, I already do that with other unit tests.

Reply:
fixed